### PR TITLE
[DSIP-24][RemoteLogging] Support AbsRemoteLogHandler

### LINKED ISSUE #15765: [DSIP-24][RemoteLogging] Add Azure Blob Storage RemoteLogHandler
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Motivation

I wants to onboard dolphinscheduler into Azure Kubernetes, which needs to do some alignment to make it fits azure stack as more as possible.

### Design Detail

1. implements RemoteLogHandler with Azure BLob Storage SDK
2. create related config.

### Compatibility, Deprecation, and Migration Plan

No compatibility work due to our good code abstraction.

### Test Plan

1.make unit test pass sonar cloud command.
2.e2e test with modifies common.properties to see if api-server/master/worker log could be passed into azure blob storage container.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pegasas**
## Purpose of the pull request

close: https://github.com/apache/dolphinscheduler/issues/15765
add azure blob remote log handler

## Brief change log

add azure blob remote log handler

## Verify this pull request

image

### CODE REVIEW

**Comment by ruanwenjun** [comment]
Can we move the properties related to the remote log to a separate config file, e.g. `remote-log.yaml`, right now, the common.properties is too large, we shouldn't add addition config into this file and the file type should be deprecated, we have a better file type like yaml.

**Comment by pegasas** [comment]
> Can we move the properties related to the remote log to a separate config file, e.g. `remote-log.yaml`, right now, the common.properties is too large, we shouldn't add addition config into this file and the file type should be deprecated, we have a better file type like yaml.

I agree with that
create an following issue for tracking this https://github.com/apache/dolphinscheduler/issues/15774

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/remote/AbsRemoteLogHandler.java
Comment (ruanwenjun) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/remote/AbsRemoteLogHandler.java]: Can we move this into the constructor?
When we get the instance by 
```
AbsRemoteLogHandler.getInstance();
```
We don't need to execute `init` since the instance has already been initialized.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/remote/AbsRemoteLogHandler.java
Comment (pegasas) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/remote/AbsRemoteLogHandler.java]: Indeed this is a solution.
Currently it follows the patterns with [GcsRemoteLogHandler , s3remotelogger and etc.](https://github.com/apache/dolphinscheduler/blob/a72505c926225989cbfdebf726a99fc1068b0b3d/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/remote/GcsRemoteLogHandler.java)
I will create another PR for fixing this.

Comment (ruanwenjun) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/remote/AbsRemoteLogHandler.java]: You don't need to follow the bad patterns, and the `init` method is not defined in the interface. You can submit another PR to fix other handler, but this handler shouldn't use `init`.

**Review by rickchengx — COMMENTED** [review]
Agree with moving the properties related to the remote log to a separate config file.
Overall LGTM, please add related UT

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/remote/AbsRemoteLogHandler.java
Comment (pegasas) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/remote/AbsRemoteLogHandler.java]: I create an PR for fixing this https://github.com/apache/dolphinscheduler/pull/15780

**Comment by pegasas** [comment]
> Agree with moving the properties related to the remote log to a separate config file. Overall LGTM, please add related UT

Thanks @rickchengx .
Update UT.
image

**Review by ruanwenjun — APPROVED** [review]
LGTM, thanks for your PR, after the CI pass I will merge into dev.

**Pull request merged**
